File: android/measure/src/main/java/sh/measure/android/RecentSession.kt
package sh.measure.android

import kotlinx.serialization.Serializable

/**
 * Information about the most recent session.
 *
 * Note that this is serialized & deserialized to shared preferences, only make backwards
 * compatible changes to this object.
 */
@Serializable
internal data class RecentSession(
    /**
     * The session id.
     */
    val id: String,
    /**
     * The epoch time when the session was created.
     */
    val createdAt: Long,
    /**
     * The epoch time when the last event was tracked.
     */
    val lastEventTime: Long = 0,
    /**
     * Whether this session crashed or not.
     */
    val crashed: Boolean = false,
    /**
     * The version code of the app for which this session was created.
     */
    val versionCode: String,
) {
    fun hasTrackedEvent(): Boolean = lastEventTime != 0L
}

File: ios/MeasureSDK/RecentSession.swift
//
//  RecentSession.swift
//  MeasureSDK
//
//  Created by Adwin Ross on 25/10/24.
//

import Foundation

struct RecentSession {
    let id: String
    let createdAt: Int64
    var lastEventTime: Int64
    var crashed: Bool
    let versionCode: String

    init(id: String,
         createdAt: Int64,
         lastEventTime: Int64 = 0,
         crashed: Bool = false,
         versionCode: String) {
        self.id = id
        self.createdAt = createdAt
        self.lastEventTime = lastEventTime
        self.crashed = crashed
        self.versionCode = versionCode
    }

    func hasTrackedEvent() -> Bool {
        return lastEventTime != 0
    }
}
